src/Behat/Config/Formatter/ProgressFormatter.php

```diff
@@ -10,12 +10,15 @@ final class ProgressFormatter extends Formatter
 
     /**
      * @param bool $timer show time and memory usage at the end of the test run
+     * @param string $showOutput show the test stdout output as part of the formatter output (yes, no, on-fail, in-summary)
```

Comment:
A question - would it be worth using a `BackedEnum` for these? (I realise might have to be two, one for Pretty & one for  Progress, since they take different options). 

I guess would have to convert that to the string value in `->toArray` for parity with YAML / CLI options but we could maybe then cast it back to the enum when we read `->formatter->getParameter()` and use it internally?

It would give us validation of the supported values (and help users to pick the right ones with the PHP config interface) and type/value-safety internally. It might also be cleaner to move the duplicated `if ($showOutput === 'yes' || ($showOutput === 'on-fail' && !$result->isPassed())) {` to the enum implementation(s) in case of any future new modes (though perhaps not, I can't at the moment think what other modes we might need).

I'd equally be very happy with this as-is if you think enums could cause issues / would be too hard to support well - just thought worth considering for new features now we are >=php8 and have the PHP config interface.

Reply:
Makes sense, thanks